feat: add C implementation for `@stdlib/stats/base/dists/frechet/logcdf` (#5517)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/dists/frechet/logcdf/README.md

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/frechet/logcdf.h"
```

#### stdlib_base_dists_frechet_logcdf( x, alpha, s, m )

> Returns the logcdf for a Fréchet distribution with input `x`, shape `alpha`, scale `s`, and location `m`.

Reviewer: ```suggestion
Evaluates the natural logarithm of the [cumulative distribution function][cdf] (CDF) for a [Fréchet][frechet-distribution] distribution with shape `alpha`, scale `s`, and location `m` at a value `x`.
```

The function accepts the following arguments:

-   **x**: `[in] double` input value.
-   **alpha**: `[in] double` shape parameter.
-   **s**: `[in] double` scale parameter.
-   **m**: `[in] double` location parameter.

```c
double stdlib_base_dists_frechet_logcdf( const double x, const double alpha, const double s, const double m );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/stats/base/dists/frechet/logcdf.h"
#include "stdlib/constants/float64/eps.h"
#include <stdlib.h>
#include <stdio.h>

static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
    double alpha;
    double x;
    double s;
    double m;
    double y;
    int i;

    for ( i = 0; i < 10; i++ ) {
        alpha = random_uniform( 0.0, 10.0 );
        x = random_uniform( 0.0, 10.0 );
        s = random_uniform( 0.0, 10.0 );

>         m = random_uniform( 0.0, 20.0 );

Reviewer: ```suggestion
        m = random_uniform( 0.0, 10.0 );
```
according to `examples/index.js`

        y = stdlib_base_dists_frechet_logcdf( x, alpha, s, m );

>         printf( "x: %lf, alpha: %lf, s: %lf, m: %lf, F(x;alpha,s,m): %lf\n", x, alpha, s, m, y );

Reviewer: ```suggestion
        printf( "x: %lf, alpha: %lf, s: %lf, m: %lf, ln(F(x;alpha,s,m)): %lf\n", x, alpha, s, m, y );
```

Reviewer: as per `examples/index.js`
